core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -100,6 +100,10 @@ public PartitionData copy() {
           found = true;
           fromProjectionPos[i] = j;
         }
+        if (fields.get(i).fieldId() == ManifestFile.SPEC_ID.fieldId()) {
```

Comment:
These modifications allow BaseFile to translate into a SparkRow with the specID as a column